src/main/java/it/pagopa/ecommerce/commons/utils/NpgApiKeyHandler.java

```diff
@@ -0,0 +1,173 @@
+package it.pagopa.ecommerce.commons.utils;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.vavr.control.Either;
+import it.pagopa.ecommerce.commons.client.NpgClient;
+import it.pagopa.ecommerce.commons.exceptions.NpgApiKeyConfigurationException;
+
+import java.util.*;
+import java.util.function.Function;
+
+/**
+ * This class handles all NPG api keys such as default one and per methods psp.
+ * Use {@link NpgApiKeyHandlerBuilder} class to initialize a new class instance
+ * api keys
+ */
+public class NpgApiKeyHandler {
```

Comment:
What abount naming this `NpgApiKeyConfiguration`?

Reply:
Fixed with https://github.com/pagopa/pagopa-ecommerce-commons/pull/188/commits/17f5fabeeddc26d002749691174e3a9fde598cf2